### PR TITLE
Add lightfm

### PR DESCRIPTION
**Pull request by rth**
This adds a setup for [lightfm](https://github.com/lyst/lightfm).

The license file is not included in the .tar.gz on PyPi, will open an issue about it upstream.
I'm not sure what's the situation with openmp support on Linux and if I need to anything particular, in addition to adding the dependency. Unless it's already included in `toolchain`.

I picked the minimal numpy versions for the build somewhat at random, I'm not sure what's the best practice on that. The `setup.py` doesn't specify the minimal version.

@maciejkula please let me know if you have any other comments, and if you are OK being added to the list of maintainers for this package. Thanks.

### CODE REVIEW

**Comment by rth** [comment]
I'm not certain why the download fails. It's fetching https://pypi.io/packages/source/l/lighfm/lighfm-1.14.tar.gz which redirects to https://files.pythonhosted.org/packages/source/l/lighfm/lighfm-1.14.tar.gz which doesn't exist.

 The download url on `pypi.org` is https://files.pythonhosted.org/packages/2f/e5/418933a214f659412ede8a9f34a69b02a2f8ff6b97221ad101d267bfd53c/lightfm-1.14.tar.gz

Should I use the hadcoded link then? I could be also missing something obvious here.

**Comment by maciejkula** [comment]
Hey @rth, this is great, thanks for doing this!

I think relatively old-ish numpy versions will work.

For the CI failures: it looks like `_lightfm_fast_openmp.so` is built correctly, but not picked up by the imports; not sure why.

**Comment by rth** [comment]
Thanks for your reponse @maciejkula .

> it looks like _lightfm_fast_openmp.so is built correctly, but not picked up by the imports; not sure why.

Yes, I'm trying to figure out what is happening in the Linux OpenMP build. Without OpenMP Mac OS passes and Windows had some numpy vs scipy compatibility issues which should now be fixed.

**Comment by rth** [comment]
For the record, the previous error for the Linux build was, 
```
ImportError: libomp.so: cannot open shared object file: No such file or directory
```
but didn't show because the import was wrapped into a `try .. except ImportError ..` clause. Adding openmp to runtime fixes it.

Currently the import still fails in, [_lightfm_fast.py](https://github.com/lyst/lightfm/blob/317d2c132a047041979c1d534cd37609542a5709/lightfm/_lightfm_fast.py#L3-L4):
```py
# Import OpenMP-enabled extension
from ._lightfm_fast_openmp import *  # NOQA
from ._lightfm_fast_openmp import __test_in_positives # NOQA
```
the first import works while the second fails. `__test_in_positives` should indeed be in that file.

**Comment by CJ-Wright** [comment]
Please restore the example `meta.yaml`

**Comment by rth** [comment]
Pinning openmp to 4.0 fixed the Linux build. All builds now pass. @CJ-Wright I also addressed your comment.

Please let me know if you have other comments. Thanks.

**Comment by maciejkula** [comment]
Again, thanks for doing this @rth! You have a [insert beverage of choice] on me if we ever meet in person.

**Comment by rth** [comment]
@maciejkula Sure that works! Thank you for developing useful and nicely written libraries )

**Comment by isuruf** [comment]
`openmp` is the llvm one. You should add `gcc` to build requirements and `libgcc` to the run requirements.

File: recipes/lightfm/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    - lightfm._lightfm_fast_openmp  # [linux]
+    - lightfm
+    - lightfm.lightfm
+
+about:
+  home: https://github.com/lyst/lightfm
+  license: Apache-2.0
+  license_family: Apache
+  # license file not included in the .tar.gz, should be fixed in the next release.
+  # license_file: LICENSE.txt
```
Comment (CJ-Wright) [inline recipes/lightfm/meta.yaml]: Can you please manually package the license? https://conda-forge.org/docs/meta.html#packaging-the-license-manually

**Comment by rth** [comment]
> openmp is the llvm one. You should add gcc to build requirements and libgcc to the run requirements.

Aww, right. This should be fixed now.  Thanks for the review.

File: recipes/lightfm/meta.yaml (hunk 1 above)
Comment (rth) [reply recipes/lightfm/meta.yaml]: Yes, done. Thanks for catching that.

**Comment by rth** [comment]
@isuruf @CJ-Wright Would you have any other comments? Thank you.

**Pull request merged**